sklearn/model_selection/_search_successive_halving.py

```diff
@@ -216,7 +220,7 @@ def _select_best_index(refit, refit_metric, results):
         """
         last_iter = np.max(results["iter"])
         last_iter_indices = np.flatnonzero(results["iter"] == last_iter)
-        best_idx = np.argmax(results["mean_test_score"][last_iter_indices])
+        best_idx = np.nanargmax(results["mean_test_score"][last_iter_indices])
```

Comment:
Maybe check for "all NaN" and handle that instead of having an exception raised by `nanargmax`

```python
if np.isnan(results["mean_test_score"][last_iter_indices]).all():
  # do something clever, random order, ??
```

Reply:
If they are all nans, then I prefer not to do anything clever. I would set `best_idx=0` and document the behavior.